ironfish-cli/src/utils/transaction.ts

```diff
@@ -14,6 +14,80 @@ import {
   UnsignedTransaction,
 } from '@ironfish/sdk'
 import { CliUx } from '@oclif/core'
+import { ProgressBar } from '../types'
+
+export class TransactionTimer {
+  private logger: Logger
+  private progressBar: ProgressBar | undefined
+  private startTime: number | undefined
+  private estimateInMs: number
+  private timer: NodeJS.Timer | undefined
+
+  constructor(
+    spendPostTime: number,
+    raw: RawTransaction,
+    logger?: Logger,
+    options: {
+      displayEstimate?: boolean
+    } = {
+      displayEstimate: true,
+    },
+  ) {
+    this.logger = logger ?? createRootLogger()
+    this.estimateInMs = Math.max(Math.ceil(spendPostTime * raw.spends.length), 1000)
+    options.displayEstimate && this.displayEstimate()
```

Comment:
Nit: it is unusual for constructors to have side-effects (in this case, printing something). I would make this behavior more explicit and require callers to call this method explicitly

Reply:
Great feedback, I had it that way originally and I'll change it back